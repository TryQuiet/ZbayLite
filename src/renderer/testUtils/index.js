import { DateTime, Settings } from 'luxon'
import secp256k1 from 'secp256k1'
import BigNumber from 'bignumber.js'
import { hash } from '../zbay/messages'
import defaultChannels from '../zcash/channels'
import { messageType } from '../../shared/static'
Settings.defaultZoneName = 'utc'

export const vaultTestMessages = [
  {
    id: '123',
    properties: {
      title: 'test',
      createdAt: '123456',
      message: '{"itemId":"6b31f1a5c68902a767eb542fa17daeb338e32d12704ac124ce55994754a5001e","text":"hello"}',
      sender: 'testsender',
      senderUsername: 'testusername',
      type: '11',
      id: '049a7d2ab30817765ea9875bfca40299669e1631329fc58b2a053ff58f78b007'
    }
  },
  {
    id: '1234',
    properties: {
      title: 'test2',
      createdAt: '123456',
      message: '{"itemId":"6b31f1a5c68902a767eb542fa17daeb338e32d12704ac124ce55994754a5001e","text":"hello"}',
      sender: 'testsender',
      senderUsername: 'testusername',
      type: '11',
      id: '149a7d2ab30817765ea9875bfca40299669e1631329fc58b2a053ff58f78b007'
    }
  }
]
export const identities = [
  {
    address: 'zs1z7rejlpsa98s2rrrfkwmaxu53e4ue0ulcrw0h4x5g8jl04tak0d3mm47vdtahatqrlkngh9slya',
    username: 'Wenus'
  },
  {
    address: 'zs1dhqp9dtr4pksnmaynp2k22qduvywejg3neqq4swd4a6gnz6w0m208kefcdm9n2067yn5clcvgsq',
    username: 'Mars'
  }
]

const privateKey = 'ceea41a1c9e91f839c96fba253b620da70992954b2a28b19322b191d8f5e56db'
const pKey = Buffer.alloc(32)
pKey.write(privateKey, 0, 'hex')

export const now = DateTime.utc(2019, 3, 7, 13, 3, 48)

export const createChannel = id => ({
  id: id,
  name: `Channel ${id}`,
  private: Boolean(id % 2),
  address: `zs1testaddress${id}`,
  unread: 12,
  description: id % 2 === 0 ? '' : `Channel about ${id}`,
  advertFee: 0.15,
  keys: {
    ivk: `incoming-viewing-key-${id}`,
    sk: 'test-secret-key'
  }
})

export const createUsersChannel = (id, network = 'testnet') => {
  return {
    id: id,
    name: `Channel ${id}`,
    private: Boolean(id % 2),
    address: defaultChannels.registeredUsers[network].address,
    unread: id,
    description: id % 2 === 0 ? '' : `Channel about ${id}`,
    keys: {
      ivk: `incoming-viewing-key-${id}`
    }
  }
}

export const channels = {
  createChannel,
  createUsersChannel
}

export const createVaultMessage = (
  id,
  createdAt = now.minus({ hours: id }).toSeconds(),
  replyTo = 'zs1z7rejlpsa98s2rrrfkwmaxu53e4ue0ulcrw0h4x5g8jl04tak0d3mm47vdtahatqrlkngh9slya'
) => ({
  id,
  status: 'broadcasted',
  spent: new BigNumber('123'),
  type: messageType.BASIC,
  createdAt,
  message: `This is some message ${id}`,
  sender: {
    replyTo,
    username: 'test'
  }
})

export const createMessage = (id, createdAt = now.minus({ hours: id }).toSeconds()) => ({
  id,
  type: messageType.BASIC,
  createdAt,
  message: `This is some message ${id}`,
  signature: secp256k1.sign(hash(`This is a message with id ${id}`), pKey).signature
})
export const createItemMessage = (id, createdAt = now.minus({ hours: id }).toSeconds()) => ({
  id,
  type: messageType.BASIC,
  createdAt,
  message: { itemId: 'test1', test: `This is some message ${id}` },
  signature: secp256k1.sign(hash(`This is a message with id ${id}`), pKey).signature
})

export const createReceivedMessage = ({
  id,
  createdAt = now.toSeconds(),
  sender = identities[0]
}) => ({
  id,
  spent: new BigNumber('0.32'),
  type: messageType.BASIC,
  createdAt,
  message: `This is a message with id ${id}`,
  signature: secp256k1.sign(hash(`This is a message with id ${id}`), pKey).signature
})

export const createSendableMessage = ({
  message,
  createdAt = now.toSeconds(),
  type = messageType.BASIC
}) => ({
  type,
  signature: secp256k1.sign(hash(message), pKey).signature,
  createdAt,
  message
})

export const createSendableUserMessage = ({
  message,
  createdAt = now.toSeconds(),
  type = messageType.USER
}) => ({
  type,
  signature: secp256k1.sign(hash(JSON.stringify(message) + createdAt), pKey).signature,
  createdAt,
  message
})

export const createSendableTransferMessage = ({
  message = 'hello',
  createdAt = now.toSeconds()
}) => ({
  type: messageType.TRANSFER,
  spent: new BigNumber('0.32'),
  sender: {
    replyTo: 'zs1z7rejlpsa98s2rrrfkwmaxu53e4ue0ulcrw0h4x5g8jl04tak0d3mm47vdtahatqrlkngh9slya',
    username: 'Wenus'
  },
  receiver: {
    replyTo: 'zs1z9rejlpsa98s2rrrfkwmaxu53e4ue0ulcrw0h4x5g8jl04tak0d3mm47vdtahatqrlkngh9slya',
    username: 'Sun'
  },
  createdAt,
  signature: secp256k1.sign(hash(message), pKey).signature,
  message
})

export const createReceivedTransferMessage = ({ id, createdAt = now.toSeconds() }) => ({
  id,
  spent: new BigNumber('0.32'),
  type: messageType.TRANSFER,
  createdAt,
  message: `This is a message with id ${id}`,
  signature: secp256k1.sign(hash(`This is a message with id ${id}`), pKey).signature
})

export const createReceivedUserMessage = ({ id, createdAt = now.toSeconds() }) => ({
  id,
  spent: new BigNumber('0.32'),
  type: messageType.USER,
  createdAt,
  message: {
    firstName: 'testname',
    lastName: 'testlastname',
    nickname: 'nickname',
    address:
      'ztestsapling14dxhlp8ps4qmrslt7pcayv8yuyx78xpkrtfhdhae52rmucgqws2zp0zwf2zu6qxjp96lzapsn4r'
  },
  signature: secp256k1.sign(
    hash(
      JSON.stringify({
        firstName: 'testname',
        lastName: 'testlastname',
        nickname: 'nickname',
        address:
          'ztestsapling14dxhlp8ps4qmrslt7pcayv8yuyx78xpkrtfhdhae52rmucgqws2zp0zwf2zu6qxjp96lzapsn4r'
      }) + id
    ),
    pKey
  ).signature,
  r: secp256k1.sign(
    hash(
      JSON.stringify({
        firstName: 'testname',
        lastName: 'testlastname',
        nickname: 'nickname',
        address:
          'ztestsapling14dxhlp8ps4qmrslt7pcayv8yuyx78xpkrtfhdhae52rmucgqws2zp0zwf2zu6qxjp96lzapsn4r'
      }) + id
    ),
    pKey
  ).recovery
})

export const messages = {
  createReceivedMessage,
  createSendableMessage,
  createReceivedTransferMessage,
  createSendableTransferMessage,
  createMessage,
  createVaultMessage,
  createReceivedUserMessage,
  createSendableUserMessage
}

export const createTransfer = ({ txid, amount = '0', memo = '', change = false, blockHeight = 100 }) => ({
  txid,
  amount,
  memo,
  memohex: memo,
  change,
  blockHeight: 100
})

export const transfers = {
  createTransfer
}

export const createIdentity = ({
  name = 'Saturn',
  address = 'saturn-private-address',
  transparentAddress = 'saturn-transparent-address',
  signerPrivKey = 'AyrXTov+4FEyle1BrxndDhePlUOB1nPV5YvyexC8m6E=',
  signerPubKey = 'Alw9G29ahXm5et9T7FJF3lbXVXNkapR826yPtfMVJYnK',
  keys = {
    tpk: 'saturn-tpk',
    sk: 'saturn-sk'
  },
  shippingData = {
    firstName: 'Saturn',
    lastName: 'the Planet',
    street: 'Coders Dv',
    country: 'Poland',
    region: 'Malopolska',
    city: 'Krakow',
    postalCode: '1337-455'
  }
} = {}) => ({
  name,
  address,
  transparentAddress,
  signerPrivKey,
  signerPubKey,
  shippingData,
  keys
})

export default {
  identities,
  now,
  transfers,
  channels,
  messages,
  createIdentity,
  createMessage,
  createSendableTransferMessage,
  createItemMessage,
  createReceivedMessage,
  vaultTestMessages
}
