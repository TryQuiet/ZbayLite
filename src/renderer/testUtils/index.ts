import { DateTime, Settings } from 'luxon'
import BigNumber from 'bignumber.js'
import { messageType } from '../../shared/static'

Settings.defaultZoneName = 'utc'

export const now = DateTime.utc(2019, 3, 7, 13, 3, 48)

export const identities = [
  {
    address: 'zs1z7rejlpsa98s2rrrfkwmaxu53e4ue0ulcrw0h4x5g8jl04tak0d3mm47vdtahatqrlkngh9slya',
    username: 'Wenus'
  },
  {
    address: 'zs1dhqp9dtr4pksnmaynp2k22qduvywejg3neqq4swd4a6gnz6w0m208kefcdm9n2067yn5clcvgsq',
    username: 'Mars'
  }
]

export const createChannel = (id: number) => ({
  id: id,
  name: `Channel ${id}`,
  private: Boolean(id % 2),
  address: `zs1testaddress${id}`,
  unread: 12,
  description: id % 2 === 0 ? '' : `Channel about ${id}`,
  advertFee: 0.15,
  keys: {
    ivk: `incoming-viewing-key-${id}`,
    sk: 'test-secret-key'
  }
})

export const channels = {
  createChannel
}

export const createVaultMessage = (
  id,
  createdAt = now.minus({ hours: id }).toSeconds(),
  replyTo = 'zs1z7rejlpsa98s2rrrfkwmaxu53e4ue0ulcrw0h4x5g8jl04tak0d3mm47vdtahatqrlkngh9slya'
) => ({
  id,
  status: 'broadcasted',
  spent: new BigNumber('123'),
  type: messageType.BASIC,
  createdAt,
  message: `This is some message ${id}`,
  sender: {
    replyTo,
    username: 'test'
  }
})

export const createMessage = async () => {
  return {
    id: '1',
    type: messageType.BASIC,
    message: 'example message',
    pubKey: 'ownPubKey',
    channelId: 'address',
    createdAt: 123445789,
    signature: 'secretsignature'
  }
}

export const messages = {
  createMessage,
  createVaultMessage
}

export const createIdentity = ({
  name = 'Saturn',
  address = 'saturn-private-address',
  transparentAddress = 'saturn-transparent-address',
  signerPrivKey = 'AyrXTov+4FEyle1BrxndDhePlUOB1nPV5YvyexC8m6E=',
  signerPubKey = 'Alw9G29ahXm5et9T7FJF3lbXVXNkapR826yPtfMVJYnK',
  keys = {
    tpk: 'saturn-tpk',
    sk: 'saturn-sk'
  },
  shippingData = {
    firstName: 'Saturn',
    lastName: 'the Planet',
    street: 'Coders Dv',
    country: 'Poland',
    region: 'Malopolska',
    city: 'Krakow',
    postalCode: '1337-455'
  }
} = {}) => ({
  name,
  address,
  transparentAddress,
  signerPrivKey,
  signerPubKey,
  shippingData,
  keys
})

export default {
  identities,
  now,
  channels,
  messages,
  createIdentity,
  createMessage
}
