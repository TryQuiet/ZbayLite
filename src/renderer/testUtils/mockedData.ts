import { IChannelInfo } from '@zbayapp/nectar'
import { Community } from '@zbayapp/nectar/lib/sagas/communities/communities.slice'
import { Identity } from '@zbayapp/nectar/lib/sagas/identity/identity.slice'
import { publicChannelsAdapter } from '@zbayapp/nectar/lib/sagas/publicChannels/publicChannels.adapter'
import { CommunityChannels } from '@zbayapp/nectar/lib/sagas/publicChannels/publicChannels.slice'

export const communityId = 'communityId'

export const community: Community = {
  id: communityId,
  name: 'name',
  CA: {
    rootCertString: 'MIIBTDCB8wIBATAKBggqhkjOPQQDAjASMRAwDgYDVQQDEwdaYmF5IENBMB4XDTEwMTIyODEwMTAxMFoXDTMwMTIyODEwMTAxMFowEjEQMA4GA1UEAxMHWmJheSBDQTBZMBMGByqGSM49AgEGCCqGSM49AwEHA0IABJWQkmtUzYuTVkEaNcA+37kGTevnCM3cxbPNTl7bpOhNBUWCcyfw4Mz//BvbRaQUM9YzXf37vSdZ0Ny7jx5io0mjPzA9MA8GA1UdEwQIMAYBAf8CAQMwCwYDVR0PBAQDAgCGMB0GA1UdJQQWMBQGCCsGAQUFBwMCBggrBgEFBQcDATAKBggqhkjOPQQDAgNIADBFAiEA05xsJ8fNTwP7T6EVDROAGpC9Z4iPlRTjlURE28D1iWoCIGT2nXMZGw9bG2ckeunoHRs0Olp5+Ike6hDPCl2jLe7P',
    rootKeyString: 'MIGTAgEAMBMGByqGSM49AgEGCCqGSM49AwEHBHkwdwIBAQQgZ3Nr6bD5eAD4+1KGNSr7HvswiPt5u9eq78vyH2L09DKgCgYIKoZIzj0DAQehRANCAASVkJJrVM2Lk1ZBGjXAPt+5Bk3r5wjN3MWzzU5e26ToTQVFgnMn8ODM//wb20WkFDPWM139+70nWdDcu48eYqNJ'
  },
  peerList: [],
  rootCa: '',
  registrarUrl: 'ugmx77q2tnm5fliyfxfeen5hsuzjtbsz44tsldui2ju7vl5xj4d447yd:7909',
  registrar: {
    privateKey: '',
    address: ''
  },
  privateKey: '',
  onionAddress: '',
  port: 0
}

export const generalChannel: IChannelInfo = {
  name: 'general',
  description: 'description',
  owner: 'holmes',
  timestamp: 1636971603355,
  address: 'general'
}

export const communityChannels = new CommunityChannels(communityId)
communityChannels.currentChannel = 'general'
communityChannels.channels = publicChannelsAdapter.setAll(
  publicChannelsAdapter.getInitialState(),
  [generalChannel]
)

export const createIdentity = (id: string = communityId): Identity => ({
  id: id,
  zbayNickname: 'holmes',
  hiddenService: {
    onionAddress: 'ugmx77q2tnm5fliyfxfeen5hsuzjtbsz44tsldui2ju7vl5xj4d447yd',
    privateKey:
        'ED25519-V3:eECPVkKQxx0SADnjaqAxheH797Q79D0DqGu8Pbc83mpfaZSujZdxqJ6r5ZwUDWCYAegWx2xNkMt7zUKXyxKOuQ=='
  },
  peerId: {
    id: 'QmPdB7oUGiDEz3oanj58Eba595H2dtNiKtW7bNTrBey5Az',
    privKey:
        'CAASqAkwggSkAgEAAoIBAQCUGW9AvS5miIuhu2xk+OiaQpaTBPDjS22KOi2KfXXFfzwyZvduO0ZsOE5HxoGQ/kqL4QR2RhbTCZ8CNdkWPDR/s8fb7JGVRLkoexLzgMNs7OFg0JFy9AjmZ/vspE6y3irr/DH3bp/qiHTWiSvOGMaws3Ma74mqUyBKfK+hIri0/1xHGWNcIyhhjMy7f/ulZCOyd+G/jPA54BI36dSprzWSxdHbpcjAJo95OID9Y4HLOWP3BeMCodzslWpkPg+F9x4XjiXoFTgfGQqi3JpWNdgWHzpAQVgOGv5DO1a+OOKxjakAnFXgmg0CnbnzQR7oIHeutizz2MSmhrrKcG5WaDyBAgMBAAECggEAXUbrwE2m9ONZdqLyMWJoNghsh+qbwbzXIDFmT4yXaa2qf2BExQPGZhDMlP5cyrKuxw0RX2DjrUWpBZ5evVdsBWZ5IXYNd4NST0G8/OsDqw5DIVQb19gF5wBlNnWCL7woMnukCOB/Dhul4x2AHo2STuanP7bQ8RrsAp4njAivZydZADv2Xo4+ll+CBquJOHRMjcIqXzaKLoXTf80euskHfizFT4cFsI6oZygx8yqstoz2SBj2Qr3hvkUmSBFhE+dChIRrpcYuuz0JPpUTBmGgCLdKarUJHH1GJ4+wc6YU9YmJJ3kqyR+h/oVGaB1j4YOd5ubtJAIvf7uj0Ofhq1FJhQKBgQDrgsrUAZCafk81HAU25EmfrvH0jbTvZ7LmM86lntov8viOUDVk31F3u+CWGP7L/UomMIiveqO8J9OpQCvK8/AgIahtcB6rYyyb7XGLBn+njfVzdg8e2S4G91USeNuugYtwgpylkotOaAZrmiLgl415UgJvhAaOf+sMzV5xLREWMwKBgQCg+9iU7rDpgx8Tcd9tf5hGCwK9sorC004ffxtMXa+nN1I+gCfQH9eypFbmVnAo6YRQS02sUr9kSfB1U4f7Hk1VH/Wu+nRJNdTfz4uV5e65dSIo3kga8aTZ8YTIlqtDwcVv0GDCxDcstpdmR3scua0p2Oq22cYrmHOBgSGgdX0mewKBgQCPm/rImoet3ZW5IfQAC+blK424/Ww2jDpn63F4Rsxvbq6oQTq93vtTksoZXPaKN1KuxOukbZlIU9TaoRnTMTrcrQmCalsZUWlTT8/r4bOX3ZWtqXEA85gAgXNrxyzWVYJMwih5QkoWLpKzrJLV9zQ6pYp8q7o/zLrs3JJZWwzPRwKBgDrHWfAfKvdICfu2k0bO1NGWSZzr6OBz+M1lQplih7U9bMknT+IdDkvK13Po0bEOemI67JRj7j/3A1ZDdp4JFWFkdvc5uWXVwvEpPaUwvDZ4/0z+xEMaQf/VwI7g/I2T3bwS0JGsxRyNWsBcjyYQ4Zoq+qBi6YmXc20wsg99doGrAoGBAIXD8SW9TNhbo3uGK0Tz7y8bdYT4M9krM53M7I62zU6yLMZLTZHX1qXjbAFhEU5wVm6Mq0m83r1fiwnTrBQbn1JBtEIaxCeQ2ZH7jWmAaAOQ2z3qrHenD41WQJBzpWh9q/tn9JKD1KiWykQDfEnMgBt9+W/g3VgAF+CnR+feX6aH',
    pubKey:
        'CAASpgIwggEiMA0GCSqGSIb3DQEBAQUAA4IBDwAwggEKAoIBAQCUGW9AvS5miIuhu2xk+OiaQpaTBPDjS22KOi2KfXXFfzwyZvduO0ZsOE5HxoGQ/kqL4QR2RhbTCZ8CNdkWPDR/s8fb7JGVRLkoexLzgMNs7OFg0JFy9AjmZ/vspE6y3irr/DH3bp/qiHTWiSvOGMaws3Ma74mqUyBKfK+hIri0/1xHGWNcIyhhjMy7f/ulZCOyd+G/jPA54BI36dSprzWSxdHbpcjAJo95OID9Y4HLOWP3BeMCodzslWpkPg+F9x4XjiXoFTgfGQqi3JpWNdgWHzpAQVgOGv5DO1a+OOKxjakAnFXgmg0CnbnzQR7oIHeutizz2MSmhrrKcG5WaDyBAgMBAAE='
  },
  userCertificate: 'userCertificate',
  dmKeys: {
    publicKey: 'publicKey',
    privateKey: 'privateKey'
  },
  userCsr: {
    userCsr: 'userCsr',
    userKey: 'userKey',
    pkcs10: {
      publicKey: 'publicKey',
      privateKey: 'privateKey',
      pkcs10: 'pkcs10'
    }
  }
})
