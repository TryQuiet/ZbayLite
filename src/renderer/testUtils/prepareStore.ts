import { communities, identity, users, errors, messages, publicChannels } from '@zbayapp/nectar'
import { StoreKeys as NectarStoreKeys } from '@zbayapp/nectar/lib/sagas/store.keys'
import { StoreKeys } from '../store/store.keys'
import { combineReducers, createStore, applyMiddleware } from 'redux'
import createSagaMiddleware from 'redux-saga'
import thunk from 'redux-thunk'
import rootSaga from '../sagas/index.saga'
import { socketActions, socketReducer } from '../sagas/socket/socket.slice'
import { modalsReducer } from '../sagas/modals/modals.slice'
import MockedSocket from 'socket.io-mock'
import { fork, delay, call, put } from 'typed-redux-saga'

export const reducers = {
  [NectarStoreKeys.Communities]: communities.reducer,
  [NectarStoreKeys.Identity]: identity.reducer,
  [NectarStoreKeys.Users]: users.reducer,
  [NectarStoreKeys.Errors]: errors.reducer,
  [NectarStoreKeys.Messages]: messages.reducer,
  [NectarStoreKeys.PublicChannels]: publicChannels.reducer,
  [StoreKeys.Socket]: socketReducer,
  [StoreKeys.Modals]: modalsReducer
}

export const prepareStore = async (
  mockedState?: { [key in StoreKeys | NectarStoreKeys]?: any },
  mockedSocket?: MockedSocket
) => {
  const combinedReducers = combineReducers(reducers)
  const sagaMiddleware = createSagaMiddleware()
  const store = createStore(
    combinedReducers,
    mockedState,
    applyMiddleware(...[sagaMiddleware, thunk])
  )
  // Fork Nectar's sagas (require mocked socket.io-client)
  if (mockedSocket) {
    sagaMiddleware.run(rootSaga)
    // Mock socket connected event
    await sagaMiddleware.run(mockSocketConnectionSaga, mockedSocket).toPromise()
  }
  return {
    store,
    runSaga: sagaMiddleware.run
  }
}

function* mockSocketConnectionSaga(socket: MockedSocket): Generator {
  yield* fork(function* (): Generator {
    yield* delay(1000)
    yield* call(() => {
      socket.socketClient.emit('connect')
    })
  })
  yield* put(socketActions.startConnection({ dataPort: 4677 }))
}
