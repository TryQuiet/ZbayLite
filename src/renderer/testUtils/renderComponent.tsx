import React, { FC, ReactElement } from 'react'
import { MuiThemeProvider } from '@material-ui/core'
import { Provider } from 'react-redux'

import { render } from '@testing-library/react'

import theme from '../theme'
import store from '../store'
import { Store } from 'redux'

export const renderComponent = (ui: ReactElement, storeState: Store = store): ReturnType<typeof render> => {
  const Wrapper: FC = ({ children }) => (
    <MuiThemeProvider theme={theme}>
      <Provider store={storeState}>
        {children}
      </Provider>
    </MuiThemeProvider>
  )

  return render(ui, { wrapper: Wrapper })
}
