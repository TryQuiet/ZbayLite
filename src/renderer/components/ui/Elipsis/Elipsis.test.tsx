import React from 'react'

import each from 'jest-each'

import { Elipsis } from './Elipsis'
import { renderComponent } from '../../../testUtils/renderComponent'

describe('Elipsis', () => {
  each(['bottom-start', 'bottom', 'bottom-end']).test(
    'renders with placement %s',
    (placement) => {
      const result = renderComponent(
        <Elipsis
          content={'this is a sample text'}
          length={5}
          tooltipPlacement={placement}
        />
      )
      expect(result.baseElement).toMatchSnapshot()
    }
  )

  it('renders with custom size', () => {
    const result = renderComponent(
      <Elipsis content={'this is a sample text'} length={5} />
    )
    expect(result.baseElement).toMatchSnapshot()
  })

  it('disables if shorter than limit', () => {
    const result = renderComponent(
      <Elipsis content={'this is a sample text'} length={50} />
    )
    expect(result.baseElement).toMatchSnapshot()
  })
})
