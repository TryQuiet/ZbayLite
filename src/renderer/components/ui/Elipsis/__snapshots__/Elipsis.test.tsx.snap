// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Elipsis renders with placement bottom 1`] = `
<body>
  <div>
    <span>
      <span
        class="MuiTypography-root makeStyles-content-48 MuiTypography-caption"
      >
        this ...
      </span>
    </span>
  </div>
</body>
`;

exports[`Elipsis renders with placement bottom-end 1`] = `
<body>
  <div>
    <span>
      <span
        class="MuiTypography-root makeStyles-content-95 MuiTypography-caption"
      >
        this ...
      </span>
    </span>
  </div>
</body>
`;

exports[`Elipsis renders with placement bottom-start 1`] = `
<body>
  <div>
    <span>
      <span
        class="MuiTypography-root makeStyles-content-1 MuiTypography-caption"
      >
        this ...
      </span>
    </span>
  </div>
</body>
`;
